overlord/snapshotstate/backend/backend.go

```diff
@@ -486,34 +492,49 @@ var ErrCannotCancel = errors.New("cannot cancel: import already finished")
 // a commit.
 type importTransaction struct {
 	id        uint64
+	flockPath string
+	flock     *osutil.FileLock
 	committed bool
 }
 
+// newImportTransaction creates a new importTransaction for the given
+// snapshot id.
 func newImportTransaction(setID uint64) *importTransaction {
-	return &importTransaction{id: setID}
+	return &importTransaction{
+		flockPath: filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_importing", setID)),
+		id:        setID,
+	}
 }
 
-func (t *importTransaction) importInProgressFilesGlob() string {
-	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_*.zip", t.id))
+// newImportTransactionFromImportFile creates a new importTransaction
+// for the given import file path. It may return an error if an
+// invalid file was specified.
+func newImportTransactionFromImportFile(p string) (*importTransaction, error) {
+	m := regexp.MustCompile("^([0-9]+)_importing$")
```

Comment:
Perhaps it would make sense to declare all the "...importing" patterns (regexp, file globs) close to each other as const at the top of the package to make sure nothing gets out of sync.